### PR TITLE
Stop ContentLocationDatabase from over-flushing

### PR DESCRIPTION
**Pull request by jbayardo**

### CODE REVIEW

File: Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs [hunk 1]
```diff
@@ -726,11 +726,10 @@ Task forceCacheFlushAsync(OperationContext context, ContentLocationDatabaseCount
                     Tracer,
                     async () =>
                     {
-                        long flushedEntries = 0;
+                        long flushedEntries = _cacheUpdatesSinceLastFlush;
```
Comment (SergeyTeplyakov) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs]: I think it deserves a comment.

**Comment by SergeyTeplyakov** [comment]
As we discussed, lets try to add a test for this logic.

**Comment by SergeyTeplyakov** [comment]
>         internal bool ForceCacheFlush(OperationContext context, ContentLocationDatabaseCounters? counter = null, bool blocking = true)

I think we should flush during shutdown.
And lets have a test there as well.

---
Refers to: Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs:691 in c71420b. [](commit_id = c71420b0b6d3e08543b11981be9e45a2100501b5, deletion_comment = False)

**Comment by jbayardo** [comment]
As discussed via Teams, flush during shutdown is unnecessary. Have added a comment to reflect that

File: Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs [hunk 2]
```diff
@@ -77,6 +77,8 @@ public abstract class ContentLocationDatabase : StartupShutdownSlimBase
         /// </summary>
         private long _cacheUpdatesSinceLastFlush = 0;
 
+        internal long CacheUpdatesSinceLastFlush => _cacheUpdatesSinceLastFlush;
```
Comment (SergeyTeplyakov) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs]: nit: nodoc or a small comment that explains that this thing is used by tests only.

File: Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs [hunk 3]
```diff
@@ -90,7 +92,7 @@ public abstract class ContentLocationDatabase : StartupShutdownSlimBase
 
         private readonly object _flushTaskLock = new object();
 
-        private Task _flushTask = BoolResult.SuccessTask;
+        internal Task FlushTask { get; private set; } = BoolResult.SuccessTask;
```
Comment (SergeyTeplyakov) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs]: The same here.

**Pull request merged**